packages/quantic/scripts/npm/check-sfdx-project.js

```diff
@@ -0,0 +1,37 @@
+const pathlib = require('path');
+const fs = require('fs');
+
+function quanticIsDependency() {
+  const upperDirectories = __dirname.split(pathlib.sep);
+  const firstUpperDirectories = upperDirectories.slice(
+    upperDirectories.length - 5
+  );
+  return (
+    firstUpperDirectories.join(pathlib.sep) ===
+    pathlib.join('node_modules', '@coveo', 'quantic', 'scripts', 'npm')
+  );
```

Comment:
I think this is a bit too complex and error prone, being based on the position of the file.

Instead, use https://nodejs.org/api/modules.html#requireresolverequest-options and trycatch the `MODULE_NOT_FOUND` error.
If the error is thrown, then return false, otherwise, after the require.resolve, return true.
Tada.

Reply:
Thanks! I'll try this👌